### ISSUE TITLE
Try to use vh / vw for fluid typography

### ISSUE THREAD

**Issue by igoradamenko** [igoradamenko/awsm.css#12]
[Source article](https://www.smashingmagazine.com/2016/05/fluid-typography/).

**Comment by vwong** [comment]
Hi,

I took a stab implementing fluid typography in my fork here https://github.com/vwong/awsm.css

In doing so, I changed the sizes of various elements, so it's quite different to your initial version.

Highlights include:

  * 1 rem = 16px below 336px, approx 49 characters per line; below this, font size is locked, so the number of characters per line drops

  * 1 rem = 20px above 600px, approx 71 characters per line; above this, the main column remains fixed, on even wider scales, the `aside` tags appear in the side; this is no different to the current behaviour

  * rem smoothly scales in between those extremes; gently preserving the ideal 49-71 characters per-line measure across a wide range of screen resolutions

  * the base leading is 1.2x (relative to the changing rem), and the main `p`, `h1`, `h2`, etc baselines (not confused with CSS `line-height`) line up, so if you had a second column, everything lines up nicely.  You can see this in the `aside` tags, for example.  [Baseliner](https://keyes.ie/things/baseliner/) is helpful to verify this.

  * header size `h1` to `h4` follow a Fibonacci scale (2, 3, 5, 8 in reverse order)

  * leading for the larger headers get smaller, so that vertical rhythm is maintained even when the headers take more than one line.  I chose this assuming that headers are short, and can tolerate smaller leading better.

Caveats, things I can't figure out, or simply can't be done in CSS:

  * `hr`, `table`, `form` disrupts vertical rhythm.  I can't figure out how to get these to line up!  They have a tendency to use OS native widgets (like scroll bars), so maybe it's a lost cause anyway

  * `samp` and `pre / code` changes fonts, and it seems that browsers get really confused when this happens

  * images and other inline media do their own thing anyway and will need Javascript hacks if we want to get to line up nicely

In short, CSS in the browser is neither InDesign nor Quark, plus it has to deal with an infinite combination of screen sizes, DPI, and user preferred zoom levels; and that's assuming the perfect browser!  Still, we can get quite a long way with some due diligence.

I'll send a pull request if you'd like to merge it to yours.  Thanks for a great project!

**Comment by igoradamenko** [comment]
@vwong, thank you for contribution to the project! I'll check your awsm changes in this weekend and tell you about my decision 🐨

**Comment by vwong** [comment]
Thanks. Feel free to ask any questions, or challenge any design decisions. I'm up for change / clarification.

For example, I'm beginning to think a base leading of 1.2x is too tight, after using it for a while; maybe 1.3x would be better.  This change is not just a simple parameter change though, due to the fact that CSS don't have any notion of baselines, and we're emulating it with line-heights + top/bottom padding.

**Comment by vwong** [comment]
@igoradamenko,

Thanks for the feedback, it's a work in progress.  And due to the subjective nature of this work, impossible to judge "perfection", even as an ideal target to work towards.

I don't know at which point you checked out my changes; I've been making a few changes, so things could be in flux.  I do take your point that 1.2x line height is too small, I made the same comment above myself.  Relaxing this would help loosen some of the other elements.

Also, I'm beginning to think that strict adherence to full baseline using CSS is not feasible (nor maybe even desirable considering the dynamic nature of the web).  Relaxing this to half-baselines may be a reasonable trade-off.

I'll play with this more, and when I'm at a point where I'm happy, I'll ping you again for more feedback.

By the way, do you consider yourself a designer with some CSS knowledge, or a programmer with some design knowledge?  I put myself in the latter.

**Comment by igoradamenko** [comment]
@vwong, I put myself in the latter too :)

**Comment by igoradamenko** [comment]
Closed by #50
